### PR TITLE
Update object models & add Realm CRUD functions

### PR DESCRIPTION
**Pull request by aoeldarrat**
### 🤔 Problem:
- bad models
- No realm network layer

### 💡 Solution:
- Updated the old models and added a new one, removed records as well
- Updated the way we're using realm, and setup CRUD functionality for Projects, Items, and Entries. Values that can be accessed using BaseService.

### 🚔 Testing:
- Im getting errors for the RealmSwift import but i think thats because of local issues with my pods. Will update when figured out

### 🗂 Task

### CODE REVIEW

File: Ledger_iOS/Models/Entry.swift [hunk 1]
```diff
@@ -5,14 +5,31 @@
...
+    @objc dynamic var id: String
+    @objc dynamic var projectId: String
+    @objc dynamic var itemId: String
+    @objc dynamic var incomeExpense: IncomeExpense
+    @objc dynamic var title: String = ""
+    @objc dynamic var desc: String = ""
+    @objc dynamic var createdDate: Date?
+    @objc dynamic var lastEditDate: Date?
+    @objc dynamic var numberOfUnits: Float
+    @objc dynamic var typeOfUnit: UnitType
```
Comment (echirchir) [inline Ledger_iOS/Models/Entry.swift:20]: Why are we switching back to the @objc dynamic anotation? @Persisted should be newer I think?

File: Ledger_iOS/Models/Item.swift [hunk 2]
```diff
@@ -5,15 +5,26 @@
...
-    @Persisted var lastEditDate: Date?
-    @Persisted var balance: Balance?
+    @objc dynamic var id: String
+    @objc dynamic var projectId: String
+    @objc dynamic var title: String?
+    @objc dynamic var desc: String?
+    @objc dynamic var entries: Array<Entry>?
+    @objc dynamic var calculatedTotal: Float = 0.0
+    @objc dynamic var createdDate: Date?
+    @objc dynamic var lastEditDate: Date?
```
Comment (echirchir) [inline Ledger_iOS/Models/Item.swift:18]: Same as above?

File: Ledger_iOS/Models/Project.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import RealmSwift
+
+class Project: Object, ObjectKeyIdentifiable {
+    @objc dynamic var id: String
+    @objc dynamic var title: String = ""
+    @objc dynamic var stringDescription: String = ""
+    @objc dynamic var items: Array<Item>?
+    @objc dynamic var createdDate: Date?
+    @objc dynamic var lastEditDate: Date?
```
Comment (echirchir) [inline Ledger_iOS/Models/Project.swift:16]: Same as above?

File: Ledger_iOS/Models/UnitType.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    case millimeter
+}
+
+@objc enum UnitType: Int {
+    case weight = 1
+    case length = 2
+    case none = 0
+}
+
+
```
Comment (echirchir) [inline Ledger_iOS/Models/UnitType.swift:32]: Unnecessary space!

File: Ledger_iOS/Models/User.swift [hunk 5]
```diff
@@ -5,14 +5,24 @@
...
 //
 
-import Foundation
 import RealmSwift
 
 class User: Object, ObjectKeyIdentifiable {
-    @Persisted(primaryKey: true) var id: ObjectId
-    @Persisted var preferredLanguage: String?
-    @Persisted var preferredCurrency: String?
-    //@Persisted var books: [Book]?
```
Comment (echirchir) [inline Ledger_iOS/Models/User.swift:15]: @Persisted is the new way I think.

File: Ledger_iOS/Models/User.swift [hunk 6]
```diff
@@ -5,14 +5,24 @@
...
+    
+    //Usually in creation order
+    @objc dynamic var id: String
+    @objc dynamic var projects: Array<Project>
```
Comment (echirchir) [inline Ledger_iOS/Models/User.swift:14]: How do we represent foreign keys in RealmSwift?

File: Ledger_iOS/Networking/BaseService.swift [hunk 7]
```diff
@@ -5,9 +5,219 @@
...
+        projectResults.map(Project.init)
+    }
+    
+    var items: [Item] {
+        itemResults.map(Item.init)
+    }
+    
+    var entries: [Entry] {
+        entryResults.map(Entry.init)
+    }
```
Comment (echirchir) [inline Ledger_iOS/Networking/BaseService.swift:38]: Very nice!

File: Ledger_iOS/Networking/BaseService.swift [hunk 8]
```diff
@@ -5,9 +5,219 @@
...
+            itemResults = readItems
+            return readItems
+        } catch let error {
+            print("error in base service: read item: \(error)")
+        }
+        return itemResults
+    }
+    
+    func getEntry() -> Results<Entry> {
+
```
Comment (echirchir) [inline Ledger_iOS/Networking/BaseService.swift:113]: White space.

**Review by echirchir — APPROVED** [review]
Looks pretty good, save for some comments

File: Ledger_iOS/Models/Entry.swift (hunk 1 above)
Comment (aoeldarrat) [reply Ledger_iOS/Models/Entry.swift:20]: After looking at the documentation you are correct, ill change them back

Comment (aoeldarrat) [reply Ledger_iOS/Models/Entry.swift:20]: Also note to self: Create additional realm models specifically for realm and models for UI

**Pull request merged**